plugins/kernelbook/skills/cuda-matmul-pipelining/SKILL.md

```
@@ -0,0 +1,171 @@
+---
+name: cuda-matmul-pipelining
+description: Implement software pipelining for CUDA matmul to overlap memory transfers with computation on Hopper+ GPUs. Use when designing multi-stage GEMM mainloops, or when hiding memory latency with async operations on sm90+.
+---
+
+# Software Pipelining for Matmul
+
+Pipelining overlaps TMA loads with WGMMA computation across multiple stages, hiding memory latency.
+
+## Pipeline Concept
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)